### DISCUSSION TITLE
Applying preset occasionally crashes otd

### DISCUSSION THREAD

**Discussion by saschamarco** [OpenTabletDriver/OpenTabletDriver#3111] [Q&A] [answered]
[Uploading log.txt…]()
I would like to use presets to change the display configuration in order to control other displays with the pen.
In about 10% of cases otd seems to crash when i apply a preset with `otd applypreset name_of_preset` and i have to restart the service with `systemctl --user restart opentabletdriver.service --now`.

There is this error in the daemon log:
`
Dez 18 02:13:16 sascha-ms7c56 otd-daemon[247655]: [Huion Kamvas Pro 24 (4K):Info]        Filters: OpenTabletDriver.Desktop.Binding.BindingHandler
Dez 18 02:13:16 sascha-ms7c56 otd-daemon[247655]: [Settings:Info]        Driver is enabled.
Dez 18 02:13:16 sascha-ms7c56 otd-daemon[247655]: [NullReferenceException:Error]        Object reference not set to an instance of an object.
Dez 18 02:13:16 sascha-ms7c56 otd-daemon[247655]:    at OpenTabletDriver.Plugin.Output.AbsoluteOutputMode.OnOutput(IDeviceReport report) in /var/tmp/pamac-build-s>
Dez 18 02:13:16 sascha-ms7c56 otd-daemon[247655]:    at OpenTabletDriver.InputDeviceTree.HandleReport(Object sender, IDeviceReport report) in /var/tmp/pamac-build>
Dez 18 02:13:16 sascha-ms7c56 otd-daemon[247655]:    at OpenTabletDriver.Devices.DeviceReader 1.Main() in /var/tmp/pamac-build-sascha/opentabletdriver/src/OpenTab>
`

In very rare cases i can not even restart otd and see this error in dmesg:
`usb 3-1.2: usbfs: USBDEVFS_CONTROL failed cmd OpenTabletDrive rqt 128 rq 6 len 255 ret -4
`
In that case i have to re-plug the usb connection and then restart otd.

This "crash" seems rarer if the pen is out of range of the display.

System info:
Manjaro unstable
Kernel 6.6.7
Huion Kamvas 24 pro 4k

Thx for any advice!

**Comment by jamesbt365** [comment, marked as answer]
Your log is poorly formatted and its quite hard to see what is going on please upload the file when you run the command if you are still encounting this issue. 
`journalctl --user-unit opentabletdriver.service > log.txt`

However, it will not be necessary to do that if you have your pen on the tablet when you are applying presets/changing settings, this is a known bug and will be handled in the next major release (0.7).
